Services/WebAccessChecker/classes/class.ilWebAccessChecker.php

```diff
@@ -144,6 +146,9 @@ public function check() {
 				return false;
 			}
 		}
+		catch (ilWACException $ex) {
+			//NOP no checking instance found or the path to the checking instance has been invalid.
```

Comment:
Is it really correct to swallow that exception here silently? Some lines below there seems to be a status change to `checked` which doesn't seem to be correct if that exception is thrown.

Reply:
@klees Thanks for the fast review!
Yes thats correct because in some cases no checking instance is registered. As far as I understand the concept correct, the web access checker distinguishes a normal and a secure path which must be checked by a registered checking instance. However if the file is not located in a secure path and no checking instance is registered the check should be successful.

I totally agree with you in terms of swallowing exceptions. But in this case there is no handling required because it simply means no checking instance found.

Reply:
right, makes sense here since the thrown ilWACException is nothing new here, but we could revamp it using a check if there is an instance available or not.

Reply:
Top, thanks a lot!